R/oc_config.R

```diff
@@ -2,30 +2,79 @@
 #'
 #' Configures session settings for \pkg{opencage}.
 #'
+#' @param key Your OpenCage API key as a character vector of length one. Do not
+#'   enter the key directly, though. See details.
 #' @param rate_sec \code{numeric(1)} Sets the maximum number of requests sent to
 #'   the OpenCage API per second. Defaults to the value set in the
 #'   \code{oc_rate_sec} option, or, in case that does not exist, to 1.
 #' @param ... Ignored.
 #'
-#' @details The rate limit allowed by the API depends on the OpenCage plan you
-#'   purchased and ranges from 1 request/sec for the "Free Trial" plan to 15
-#'   requests/sec for the "Medium" or "Large plan, see
-#'   \url{https://opencagedata.com/pricing} for details and up-to-date
-#'   information. You can set the rate limit persistently across sessions by
-#'   setting the \code{oc_rate_sec} \link[base:options]{option} in your
-#'   \code{\link[base:Startup]{.Rprofile}}. If you have the \code{usethis}
-#'   package installed, you can edit your \code{\link[base:Startup]{.Rprofile}}
-#'   most easily with \code{usethis::edit_r_profile()}.
+#' @section Configure your OpenCage API key:
+#'
+#' opencage uses the environment variable \code{OPENCAGE_KEY} internally to
+#' retrieve your OpenCage key. \code{\link{oc_config}} will help to set that
+#' environment variable. Do not set the key directly, though, as you risk
+#' exposing the key via your skript or your history. Instead, if set to
+#' \code{NULL}, the default, \code{\link{oc_config}} will check if the
+#' environment variable \code{OPENCAGE_KEY} is already set, and prompt you to
+#' enter the key in the console if it is not.
+#'
+#' If you use a package like keyring, you can safely set your key in a script
+#' like this \code{oc_config(key = keyring::key_get("opencage"))}.
+#'
+#' Finally you can save your API key directly as an environment variable in
+#' `.Renviron` as described in
+#' \href{https://happygitwithr.com/github-pat.html#step-by-step}{Happy Git and
+#' GitHub for the useR} or
+#' \href{https://csgillespie.github.io/efficientR/set-up.html#renviron}{Efficient
```

Comment:
I was thinking it might be useful to mention one could have several OpenCage keys by e.g. using different key rings?

Not used but maybe useful to mention (although I suppose the docs of this package can't explain all these things :-D): https://cran.r-project.org/web/packages/startup/

Reply:
Re multiple keys/different key rings: That sounds more of a topic for keyring. While I do have two OC keys myself (company & private), I don't think this is a common (or complicated to solve) problem. I also don't want to endorse the use of multiple free accounts.

Re {startup}: I view this as beyond scope for these docs and more fitting for a general "where to store API keys" text. TBH I am not convinced of the usefulness of {startup} for this task (with all due respect for Henrik's work).

Reply:
👍